protocol-units/settlement/mcr/contracts/script/helpers/Helper.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+contract Helper {
+    // string to address
+    function s2a(bytes memory str) public returns (address addr) {
```

Comment:
Did you get this from somewhere in particular? I think you may want to use the free memory pointer instead, i.e., `0x40`.

Reply:
https://stackoverflow.com/questions/73540193/what-happens-exactly-when-you-assign-mload0x40-to-an-address-variable

Reply:
More in depth discussion here: https://forum.soliditylang.org/t/non-memory-safe-assembly/2368/6

0 is in the scratch space, but I don't believe the scratch space resets after each assembly block. Hence 0x40 might be safer.

Reply:
You can actually just do this on the stack too...

```sol
bytes32 data = keccak256(str);
assembly {
    addr := data
}
```